scikeras/wrappers.py

```diff
@@ -755,20 +797,24 @@ def _initialize(
             # int or None
             self._random_state = self.random_state
 
-        X, y = self._validate_data(X, y, reset=True)
+        X, y, sample_weight = self._validate_data(X, y, sample_weight, reset=True)
 
         self.target_encoder_ = self.target_encoder.fit(y)
         target_metadata = getattr(self.target_encoder_, "get_metadata", dict)()
         vars(self).update(**target_metadata)
         self.feature_encoder_ = self.feature_encoder.fit(X)
-        feature_meta = getattr(self.feature_encoder, "get_metadata", dict)()
+        feature_meta = getattr(self.feature_encoder_, "get_metadata", dict)()
         vars(self).update(**feature_meta)
 
         self.model_ = self._build_keras_model()
 
-        return X, y
+        self.dataset_transformer_ = self.dataset_transformer.fit((X, y, sample_weight))
+        dataset_meta = getattr(self.dataset_transformer_, "get_metadata", dict)()
+        vars(self).update(**dataset_meta)
```

Comment:
:-1:

Reply:
```suggestion
        self.dataset_meta_ = dataset_meta
```

Reply:
Which part of these 3 lines are you 👎 on? This code hasn't stayed up to date with our discussions, so I'm not sure if it's the tuple part or something else.

Reply:
Ah, sorry, I see now we cross posted.

Reply:
I think this comment is tangential to the other discussion. I would rather consolidate similar keys into one dictionary. I'm not a fan of blindly assigning attributes to objects.

Reply:
Agreed. We actually weren't using this at the moment, and there's no reason it can't be added at a later date if a good feature requires it, so I was able to remove it for now. Does that sound like it resolves this problem?

Reply:
Yeah, I think so. I'm not opposed to setting the metadata about the dataset in a single attribute:

``` python
self.dataset_meta_ = getattr(self.dataset_transformer_, "get_metadata", dict)()
```

Reply:
Since I removed calling `get_metadata` for `dataset_transformer_`, I'm going to resolve this comment.